cgpsa

```diff
@@ -2909,30 +3015,73 @@ sub readConfigFile
     {
       $conf->{sa_local_rules_dir} = $1;
     }
-    elsif ($line =~ /^absolute[-_]queue[-_]filenames[\s\=]+(\S+)$/) 
+    elsif ($line =~ /^absolute[-_]queue[-_]filenames[\s\=]+(\S+)$/)
     {
       $conf->{absolute_queue_filenames} = evaluateBoolean($1);
     }
     elsif ($line =~ /^prevent[-_]backscatter[\s\=]+(\S+)$/)
     {
       $conf->{prevent_backscatter} = evaluateBoolean($1);
     }
+    elsif ($line =~ /^x[-_]junk[-_]score[-_]header[\s\=]+(\S+)$/)
+    {
+      $conf->{x_junk_score_header} = evaluateBoolean($1);
+    }
+    elsif ($line =~ /^x[-_]junk[-_]score[-_]thresholds[\s\=]+(\S+)$/)
+    {
+      # got to parse a space-separated list of floating point numbers;
+      # if there are not 6 of them, or they are not increasing, we
+      # keep the previous settings (which might be defaults)
+
+      my @new_x_junk_score_thresholds = split /\s+/, $1;
+      my $legit_thresholds = 1;
+      my $last_threshold = -inf;
+      foreach my $threshold (@x_junk_score_thresholds)
+      {
+        $legit_thresholds =
+          $legit_thresholds && isFloat($threshold) && $last_threshold < $threshold;
+        $last_threshold = $threshold;
+      }
+      if ($legit_thresholds)
+      {
+        @x_junk_score_thresholds = @new_x_junk_score_thresholds;
+        $conf->{x_junk_score_thresholds} = 1;
+      }
+      else
+      {
+        # flag that we read invalid thresholds
+
+        $conf->{x_junk_score_thresholds} = 0;
```

Comment:
I realized this is used to signal an invalid header, but since in the initial version it was not checked, I just removed it. However, since the value does not correspond to the value in the config file, it may be a bit confusing. How about naming it `valid_x_junk_score_thresholds`

Reply:
Sounds likg a good idea.